Maybe some useful text tweaks? (#1)
davatron5000/htmlwithsuperpowers: docs/get-started/elements.md

# Template and Slot Elements

> There are two elements which can be very useful in creating Web Components which you might not be familliar with: `<template>` and `<slot>`, for creating content fragments and passing content in to components.

Reviewer: Your correction didn't read well to me and had a typo. Something like this perhaps?

> Two handy elements for creating content fragments and passing content to components you might not be familiar with are `<template>` and `<slot>`.

Author: sure

## `<template>`

The `template` element is for creating "special" content fragments for reuse later in the application. 
By default, you won't see the contents of templates rendered in the page.  Templates are often used to 
define the Shadow Tree for a component.

```html
<template>
  <div> HTML goes here </div>
</template>
```

> The superpower :sparkles: of the `<template>` element is that it gets parsed by the HTML parser into a reusable and "inert" content fragment.  That is, the browser will not attempt to do things like load replaced elements or execute scripts inside the template.  The fragement created and be worked with as a parsed tree and inserted into the DOM relatively cheaply.

Reviewer 2: nit: typo, though that last sentence looks wrong (from a non-native PoV)

```suggestion
The superpower :sparkles: of the `<template>` element is that it gets parsed by the HTML parser into a reusable and "inert" content fragment.  That is, the browser will not attempt to do things like load replaced elements or execute scripts inside the template.  The fragment created and be worked with as a parsed tree and inserted into the DOM relatively cheaply.
```

Author: Oh wow, thanks - I fixed that last sentence as a whole, I think it was a copy/paste editing flub - please let me know if that reads better

Reviewer 2: Much better yes :wink:

## `<slot>`

> The `<slot>` element is for use inside a Shadow tree.  It provides a way to expose a "hole" into the Shadow DOM which the consumer can use to provide HTML content ("Light DOM") which will be "projected".   If you've ever used Vue's slots feature this should be familiar.

Reviewer 3: One issue I have is we're jumping to using Shadow DOM to explain something before that concept has been introduced in the guidebook. Learners with zero knowledge will have no idea what "Shadow tree" is. Those terms are pretty opaque from the outside. I'd also like to remove some of the passive voice. 

```suggestion
The `<slot>` element creates a "hole" inside a web component's [Shadow DOM](/get-started/shadow-dom.html). We explain Shadow DOM later, but for now understand slots are a way to project your custom HTML content inside a web component. Not all web components have slots and Shadow DOM, but they are very common and one of web components' greatest superpowers :sparkles: 

If you've ever used Vue's slots feature this should be familiar.
```

Author: I feel like that can still be improved, but I'm not sure how and what you've suggested seems much better than what's there now so I'm going to just commit it.

> The `<slot>` element is for use inside a Shadow tree.  It provides a way to expose a "hole" into the Shadow DOM which the consumer can use to provide HTML content ("Light DOM") which will be "projected".   If you've ever used Vue's slots feature this should be familiar.

Reviewer 2: nit: consistent capitalization
```suggestion
The `<slot>` element is for use inside a Shadow Tree.  It provides a way to expose a "hole" into the Shadow DOM which the consumer can use to provide HTML content ("Light DOM") which will be "projected".   If you've ever used Vue's slots feature this should be familiar.
```

Author: I know it _feels_ like it should be, but it really is little 't' https://dom.spec.whatwg.org/#shadow-trees 
https://developer.mozilla.org/en-US/docs/Glossary/Shadow_tree

I didn't see I had flipped it, so it's the other one that is incorrect. Fixed there.
